core/src/main/java/org/apache/brooklyn/feed/http/HttpFeed.java

```diff
@@ -407,14 +410,26 @@ private HttpToolResponse createHttpToolRespose(HttpResponse response) throws IOE
         Map<String,? extends List<String>> headers = (Map<String, List<String>>) (Map<?, ?>) response.headers().asMap();
 
         byte[] content = null;
-        if (response.getContentLength() > 0) {
-            content = ByteStreams.toByteArray(response.getContent());
+        final long durationMillisOfFirstResponse;
+        final long durationMillisOfFullContent;
+        final long startTime = System.currentTimeMillis();
+
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        durationMillisOfFirstResponse = Duration.sinceUtc(startTime).toMilliseconds();
+        try {
+            ByteStreams.copy(response.getContent(), out);
+            content = out.toByteArray();
+        } catch (IOException e) {
+            throw Throwables.propagate(e);
```

Comment:
In Brooklyn the recommended is `Exceptions.propagate`, but this works as well in this case.